src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift

```diff
@@ -105,68 +107,52 @@ extension RiskProvider: RiskProviding {
 	}
 
 	/// Called by consumers to request the risk level. This method triggers the risk level process.
-	func requestRisk(userInitiated: Bool, completion: Completion? = nil) {
+	func requestRisk(userInitiated: Bool, ignoreCachedSummary: Bool = false, completion: Completion? = nil) {
 		queue.async {
-			self._requestRiskLevel(userInitiated: userInitiated, completion: completion)
+			self._requestRiskLevel(userInitiated: userInitiated, ignoreCachedSummary: ignoreCachedSummary, completion: completion)
 		}
 	}
 
-	private struct Summaries {
-		var previous: SummaryMetadata?
-		var current: SummaryMetadata?
-	}
-
-	private func determineSummaries(
+	private func determineSummary(
 		userInitiated: Bool,
-		completion: @escaping (Summaries) -> Void
+		ignoreCachedSummary: Bool = false,
+		appConfiguration: SAP_ApplicationConfiguration,
+		completion: @escaping (SummaryMetadata?) -> Void
 	) {
 		Log.info("RiskProvider: Determine summeries.", log: .riskDetection)
-
-		// Here we are in automatic mode and thus we have to check the validity of the current summary
-		let enoughTimeHasPassed = configuration.shouldPerformExposureDetection(
-			activeTracingHours: store.tracingStatusHistory.activeTracing().inHours,
-			lastExposureDetectionDate: store.summary?.date
-		)
-		if !enoughTimeHasPassed || !self.exposureManagerState.isGood {
-			completion(
-				.init(
-					previous: nil,
-					current: store.summary
-				)
+		if !ignoreCachedSummary {
```

Comment:
How do we make sure that we don't run into the rate-limit?

Reply:
Because we assume an app configuration change is a rare event, and we are currently not using the api close to it's rate limit, we don't take any precautions here. We just give it a try, and if it fails, we keep the old summary until the next detection happens by itself.

Reply:
Okay makes sense, thanks for the explanation 👍